### PR TITLE
core: (parser) factor out parse_regions

### PR DESCRIPTION
**Pull request by tobiasgrosser**

### CODE REVIEW

**Review by georgebisbas — APPROVED** [review]

File: xdsl/parser/core.py [hunk 1]
```diff
@@ -634,6 +634,14 @@ def parse_region(self, arguments: Iterable[Argument] | None = None) -> Region:
             self.raise_error("Expected region!")
         return region
 
+    def parse_regions(self) -> list[Region]:
```
Comment (math-fehr) [inline xdsl/parser/core.py:637]: Can you rename `parse_regions` to `parse_region_list`? To make it consistent with MLIR.
Also, can you add documentation like above on the syntax you are parsing? So it is clear for users that you are parsing `(` and `)` as well if present, and that you are returning `[]` if nothing was parsed.

**Review by math-fehr — APPROVED** [review]
Just added a comment on the name and the documentation, but otherwise it's good to me!

File: xdsl/parser/core.py (hunk 1 above)
Comment (kingiler) [reply xdsl/parser/core.py]: I have updated the name and the doc. I'm not sure the parse format is correct or not?

File: xdsl/parser/core.py [hunk 2]
```diff
@@ -634,6 +634,19 @@ def parse_region(self, arguments: Iterable[Argument] | None = None) -> Region:
             self.raise_error("Expected region!")
         return region
 
+    def parse_region_list(self) -> list[Region]:
+        """
+        Parse the list of operation regions.
+        If no regions are present, returns an empty list.
```
[the remark this answers is not in the document]
Comment (kingiler) [reply xdsl/parser/core.py:640]: Okay, I'm just blindly copying from `_parse_op_result_list` doc.

File: xdsl/parser/core.py [hunk 3]
```diff
@@ -764,7 +777,7 @@ def _parse_op_result(self) -> tuple[Span, int]:
     def _parse_op_result_list(self) -> list[tuple[Span, int]]:
         """
         Parse the list of operation results.
-        If no results are present, returns an empty list.
+        If no results are present, return an empty list.
```
Comment (tobiasgrosser) [inline xdsl/parser/core.py:780]: Can you drop this change, as this is unrelated.

Comment (kingiler) [reply xdsl/parser/core.py]: Sorry for introducing unrelated changes.
